samples/apps/bookinfo/CONSUL.md

```diff
@@ -0,0 +1,76 @@
+# Consul Adapter for Istio on Docker
+
+Make Istio run in docker environment by integrating Consul as a service registry.
+
+## Design Principle
+
+The key issue is how to implement the ServiceDiscovery interface functions in Istio. 
+This platform adapter uses Consul Server to help Istio monitor service instances running in the underlying platform.
+When a service instance is brought up in docker, the [Registrator](http://gliderlabs.github.io/registrator/latest/)
+automatically registers the service in Consul.
+
+Noted that Istio pilot is running inside each app container so as to coordinate Envoy and the service mesh.
+
+## Prequisites
+
+ * Clone Istio Pilot [repo](https://github.com/istio/pilot) (required only if building images locally)
+
+ * Download and install Kubernetes CLI [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version
+  1.7.3 or higher.
+ 
+ * Download istioctl from Istio's [releases page](https://github.com/istio/istio/releases) or build from 
+ source in Istio Pilot repository
+
+## Bookinfo Demo
+
+The ingress controller is still under construction, rounting functionalities can be tested by curling a service container directly.
+
+First step is to configure kubectl to use the apiserver created in the steps below:
+ 
+```
+kubectl config set-cluster local --server=http://172.28.0.13:8080
+kubectl config set-context local --cluster=local
+kubectl config use-context local
+```
+
+To bring up all containers directly, from the `/src` directory run 
+
+    docker-compose up -d 
+
+This will pull images from docker hub to your local computing space.
+
+Now you can see all the containers in the mesh by running `docker ps -a`.
+
+To view the productpage webpage, retrieve the productpage container IP using the `docker inspect`command:
+
+```
+$ docker inspect -f '{{.NetworkSettings.Networks.src_envoymesh.IPAddress}}' src_productpage-v1_1
+172.28.0.6
+``` 
+ 
+Open a web browser and enter the productpage IP as follows: `<IP>:9080/productpage`
+
+If you refresh the page several times, you should see different versions of reviews shown in productpage 
+presented in a round robin style (red stars, black stars, no stars). If the webpage is not displaying properly,
+you may need to run `docker-compose restart discovery` to resolve a timing issue during start up.
+
+You can create basic routing rules using istioctl from the `/bookinfo` directory:
+
+```
+istioctl create -f consul-reviews-v1.yaml
```

Comment:
Rules have been moved to /rules directory. It would be good to place them there from the beggining.

Reply:
I suggest bookinfo/consul, or maybe bookinfo/consul/rules

Reply:
@andraxylia not all rules have been moved to /rules.  There are duplicates in both directories.  the rules in /rules can be used by kubectl only, when I tried yesterday they did not work with istioctl

```
$ istioctl create -f ../rules/route-rule-all-v1.yaml 
Error: cannot parse proto message: unknown spec type 
```

conversation yesterday in #pilot

```
Laurent Demailly [1:54 PM] 
afaik istioctl understand bookinfo/ and kubetcl understands bookinfo/rules/

[1:55] 
ie istioctl doesn't understand v2

kuat
[1:55 PM] 
istioctl understands v2 spec but v1 meta
```

Reply:
We need to do something about this, like moving the old rules under a directory v0.1 or similar. Having them at the top level would be very confusing. Of course, this can be done separate from your PR, but in your PR I would refer to the new rules and use kubectl, if istioctl has not been fixed to understand v2.

Reply:
The rule files in bookinfo are obsolete v1 - they won't work with istioctl or kubectl. They will be removed as soon as we figure out how to fix broken links in blog. The rules in /rules are the new ones, currently only work with kubectl, until istioctl is changed to use the same format as kubectl.

Your rules in this PR seem to written using the partial-v2 istioctl-compatible format, which will work for now, but once istioctl is updated to support the proper new format, they will need to be changed.

So, I think you are OK to put your rules in /consul/rules and use istioctl to create them for now.